FWCore/Framework/src/EventSelector.cc

```diff
@@ -72,26 +72,20 @@ namespace edm {
         nPathNames_(0) {}
 
   EventSelector::EventSelector(ParameterSet const& config, Strings const& pathNames)
-      : pathspecs_(config.empty() ? Strings() : initPathSpecs(config.getParameter<Strings>("SelectEvents"))),
-        results_from_current_process_(true),
-        accept_all_(initAcceptAll()),
-        absolute_acceptors_(),
-        conditional_acceptors_(),
-        exception_acceptors_(),
-        all_must_fail_(),
-        all_must_fail_noex_(),
-        psetID_(),
-        nPathNames_(0) {
-    initPathNames(pathNames);
-  }
+      : EventSelector(
+            [&config]() {
+              auto const& pset = config.getUntrackedParameterSet("SelectEvents");
+              return (pset.empty() ? Strings() : pset.getParameter<Strings>("SelectEvents"));
+            }(),
+            pathNames) {}
```

Comment:
I tentatively added ecb204af3ea, because I noticed that the PSet-based constructor of `EventSelector` is somewhat inconsistent with its `fillDescription` method (the latter defines `SelectEvents.SelectEvents`, while the ctor expects `SelectEvents`). This required small changes to 3 `FWCore` unit tests. With this PR, the PSet-based ctor of `EventSelector` would not be used anywhere, I think (outside those unit tests), so it could even be removed. To be clear, ecb204af3ea is not necessary for this PR, so I could remove it if preferred.

Reply:
Sounds good. I will remove https://github.com/cms-sw/cmssw/commit/ecb204af3eaff0c652280e879a40ba0983f82e05 from this PR, to keep it 'DQM only' (although I could use review comments on this PR too, e.g. https://github.com/cms-sw/cmssw/pull/40420#discussion_r1062572628), and I'll open a separate PR for the small changes in `FWCore`.